demo-data-generator/package.yaml

```diff
@@ -0,0 +1,84 @@
+id: demo-data-generator
+name: Demo Data Generator
+author: Tenzir
+author_icon: https://raw.githubusercontent.com/tenzir/library/main/author.svg
+package_icon: https://raw.githubusercontent.com/tenzir/library/main/demo-data-generator/package.svg
+
+description: |
+  This package provides a live Zeek and Suricata event feed for demo purposes.
+categories:
+- sources
+
+inputs:
+  speed:
+    name: Speed
+    description: |
+      A speed of 1.0 corresponds to a data velocity convenient for most demos.
+    type: string
+    default: 1.0
+
+pipelines:
+  zeek:
+    name: Zeek Demo Data
+    description: |
+      Publishes a live demo data feed to the `zeek` topic.
+    restart-on-error: 1min
+    definition: |
+      let $speed = {{ inputs.speed }}
+      let $first = 2021-11-18T10:00:00Z
+      let $last = 2021-11-19T01:00:00Z
+      let $default = 3h
+      let $period = $default / $speed
+      let $factor = ($last - $first) / $period
+
+      every $period {
+        from {
+          url: "https://storage.googleapis.com/tenzir-datasets/M57/zeek-all.log.zst",
+        }
+      }
+      http url {
+        decompress_zstd
+        read_zeek_tsv
+        where $first <= ts and ts <= $last
+        sort ts
```

Comment:
Is this needed?

I understand where you are coming from, but many tools that produce high-velocity data in realtime do not have sorted timestamps. Zeek is notorious for this because it summarizes protocol activity often in parallel, resulting only quasi-sorted log streams.

Reply:
Also I think it's just nicer for demo purposes to have that steady stream.

Reply:
Gotcha. For a finite demo, I think this is acceptable.

For a continuous stream, it's obviously not really possible, which is why I was asking in the first place.

I'm resolving this.